service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/create/CreateGcsBucketStep.java

```diff
@@ -90,6 +92,10 @@ public StepResult doStep(FlightContext flightContext)
         throw new DuplicateResourceException(
             "The provided bucket name is already in use, please choose another.", storageException);
       }
+      if (storageException.getCode() == HttpStatus.SC_BAD_REQUEST) {
+        throw new InvalidReferenceException(
+            "The provided bucket name is invalid, please follow the naming instruction on https://cloud.google.com/storage/docs/naming-buckets#requirements");
```

Comment:
I think a `BAD_REQUEST` may not always mean an invalid bucket name. Another property (e.g. location) may also be the problem.

Also, I think it would be better to wrap the `StorageException` in the `InvalidReferenceException` so that we have a cause and can log that original excpetion.

Reply:
Changed the error message and wrap the StorageException. PTAL